lnwire/fuzz_test.go

```diff
@@ -822,3 +822,160 @@ func FuzzCustomMessage(f *testing.F) {
 		harness(t, data)
 	})
 }
+
+// prefixWithFailCode adds a failure code prefix to data.
+func prefixWithFailCode(data []byte, code FailCode) []byte {
+	var codeBytes [2]byte
+	binary.BigEndian.PutUint16(codeBytes[:], uint16(code))
+	data = append(codeBytes[:], data...)
+
+	return data
+}
+
+// onionFailureHarness performs the actual fuzz testing of the appropriate onion
+// failure message. This function will check that the passed-in message passes
+// wire length checks, is a valid message once deserialized, and passes a
+// sequence of serialization and deserialization checks.
+func onionFailureHarness(t *testing.T, data []byte, code FailCode) {
+	data = prefixWithFailCode(data, code)
+
+	// Don't waste time fuzzing messages larger than we'll ever accept.
+	if len(data) > MaxSliceLength {
+		return
+	}
+
+	// First check whether the failure message can be decoded.
+	r := bytes.NewReader(data)
+	msg, err := DecodeFailureMessage(r, 0)
```

Comment:
in the future we could pass in a version integer instead of 0 here

Reply:
If/when that time comes, it should be straightforward to add a version param to the fuzz tests and migrate the existing corpus.